src/main/java/com/truelayer/java/http/auth/cache/ICredentialsCache.java

```diff
@@ -1,28 +1,13 @@
 package com.truelayer.java.http.auth.cache;
 
 import com.truelayer.java.auth.entities.AccessToken;
-import com.truelayer.java.entities.RequestScopes;
 import java.util.Optional;
 
 public interface ICredentialsCache {
 
-    /**
-     * Gets the cached access token for the given request scopes.
-     * @param scopes the requested scopes
-     * @return an optional access token. If the token is expired an empty optional is returned
-     */
-    Optional<AccessToken> getToken(RequestScopes scopes);
+    Optional<AccessToken> getToken(String key);
```

Comment:
This is the main change: users of a custom cache won't have to bother computing hash keys. They will receive what the client builds, and they have the option to customise/override those if needed